types/part_set.go

```diff
@@ -373,3 +377,28 @@ func (ps *PartSet) MarshalJSON() ([]byte, error) {
 		ps.partsBitArray,
 	})
 }
+
+type RowSet struct {
+	DAHeader *ipld.DataAvailabilityHeader
+
+	rows     [][]byte
+	rowsHave *bits.BitArray
+}
```

Comment:
might need to eventually add an additional mutex here. Not something to add now, just something to think about.